src/popup/router/components/TransferReview.vue

```diff
@@ -141,102 +141,97 @@ export default {
+      if (this.isAddressUrl) {
+        this.sendTip({
+          amount,
+          recipient,
           selectedAsset,
           note,
-        } = this.transferData;
+        });
+      } else {
+        this.transfer({
+          amount,
+          recipient,
+          selectedAsset,
+        });
+      }
+    },
+    async transfer({ amount, recipient, selectedAsset }) {
+      this.loading = true;
+      try {
+        let actionResult;
 
-        if (!amountRaw || !recipient || !selectedAsset) {
-          return;
+        if (this.transferData.invoiceId != null) {
+          actionResult = await this.$store.dispatch('fungibleTokens/burnTriggerPoS', [
+            selectedAsset.contractId,
+            amount,
+            this.transferData.invoiceContract,
+            this.transferData.invoiceId,
+            { waitMined: false, modal: false },
+          ]);
+        } else if (selectedAsset.contractId !== AETERNITY_CONTRACT_ID) {
+          actionResult = await this.$store.dispatch('fungibleTokens/transfer', [
+            selectedAsset.contractId,
+            recipient,
+            amount,
+            { waitMined: false, modal: false },
+          ]);
+        } else {
+          actionResult = await this.sdk.spend(amount, recipient, {
+            waitMined: false,
+            modal: false,
+          });
         }
 
-        const amount = (selectedAsset.contractId === AETERNITY_CONTRACT_ID)
-          ? aeToAettos(amountRaw)
-          : convertToken(amountRaw, selectedAsset.decimals);
-
-        if (this.isAddressUrl) {
-          this.sendTip({
+        if (actionResult && selectedAsset.contractId !== AETERNITY_CONTRACT_ID) {
+          this.$store.dispatch('addPendingTransaction', {
             amount,
             recipient,
-            selectedAsset,
-            note,
+            hash: actionResult.hash,
+            type: 'spendToken',
+            pendingTokenTx: true,
+            tx: {
+              callerId: this.account.address,
+              contractId: selectedAsset.contractId,
+              type: SCHEMA.TX_TYPE.contractCall,
+              function: 'transfer',
+            },
           });
-        } else {
-          this.transfer({
+        } else if (actionResult) {
+          this.$store.dispatch('addPendingTransaction', {
+            hash: actionResult.hash,
             amount,
-            recipient,
-            selectedAsset,
+            type: 'spend',
+            tx: {
+              senderId: this.account.address,
+              recipientId: recipient,
+              type: SCHEMA.TX_TYPE.spend,
+            },
           });
         }
       } catch (e) {
-        this.$store.dispatch('modals/open', {
-          name: MODAL_DEFAULT,
-          title: this.$t('modals.transaction-failed.msg'),
-          icon: 'critical',
-        });
+        this.openTransactionFailedModal();
         throw e;
       } finally {
         this.loading = false;
+        this.$emit('success');
```

Comment:
should we emit "success" even if the error was thrown?

Reply:
You have right, it should not. Fixed...